crates/rustc_codegen_spirv/src/codegen_cx/entry.rs

```diff
@@ -37,9 +40,27 @@ impl<'tcx> CodegenCx<'tcx> {
         };
         let fn_hir_id = self.tcx.hir().local_def_id_to_hir_id(local_id);
         let body = self.tcx.hir().body(self.tcx.hir().body_owned_by(fn_hir_id));
+        const EMPTY: ArgAttribute = ArgAttribute::empty();
         for (abi, arg) in fn_abi.args.iter().zip(body.params) {
             match abi.mode {
-                PassMode::Direct(_) | PassMode::Indirect { .. } => {}
+                PassMode::Direct(_)
+                | PassMode::Indirect { .. }
+                // plain DST/RTA/VLA
+                | PassMode::Pair(
+                    ArgAttributes {
+                        pointee_size: Size::ZERO,
+                        ..
+                    },
+                    ArgAttributes { regular: EMPTY, .. },
+                )
+                // DST struct with fields before the DST member
+                | PassMode::Pair(
+                    ArgAttributes { .. },
+                    ArgAttributes {
+                        pointee_size: Size::ZERO,
+                        ..
+                    },
```

Comment:
The second element in a pair for `&SomeUnsizedType` is always the "metadata" like `usize` for anything based on slices. I assume `pointee_size` is just zero for non-pointers (and I'm not sure checking `pointee_size` even makes sense?).

You could probably just check that the type of the `PassMode::Pair` argument is `TyKind::Ref`, that should limit it to `&SomeUnsizedType`.